pkg/codegen/nodejs/gen.go

```diff
@@ -112,9 +113,12 @@ func (mod *modContext) tokenToModName(tok string) string {
 	return modName
 }
 
-func (mod *modContext) tokenToType(tok string, input bool) string {
+func (mod *modContext) tokenToType(tok string, input, enum bool) string {
 	modName, name := mod.tokenToModName(tok), tokenToName(tok)
 
+	if enum {
+		return "enums." + modName + title(name)
+	}
```

Comment:
Do we need to emit enums into their own namespace, or can they live in the same namespace as resources?

Reply:
Yeah, I was asking myself the same question. Why do inputs and outputs live in their own namespaces? Naming conflicts?

Reply:
There are a lot of references to enums in the azure code. Object types and resources have enums as their properties, meaning there's a lot of importing enums and then referencing the types. So I think they do need to be namespaced, otherwise I'm not sure how those imports would work.

Reply:
It's not great that the import path is now, for example, `aws.types.enums.lambda.NodeJS10dXRuntime` instead of `aws.lambda.NodeJS10dXRuntime`. 

I'm trying to figure out if there's some way to export as an alias from within the specific module, but I can't quite work out the semantics or whether that's possible. For example, I can `export {lambda} from "../types/enums";` in `lambda/index.ts`, but that would make the enum path `aws.lambda.lambda.NodeJS10dXRuntime`. What I really need is something like `export {* from lambda} from "../types/enums";` but that's not a real thing. Also not clear how this would work for provider-level enums.

I'm wondering if it makes sense to have `types/enums` follow a nested file module structure, so we can `export * from "../types/enums/lambda";` in `lambda/index.ts`, which would allow us to maintain the `aws.lambda.NodeJS10dXRuntime` enum path.

Reply:
I haven't looked at all the details here - but my sense is that `aws.lambda.NodeJS10dXRuntime` is *much* better than `aws.types.enums.lambda.NodeJS10dXRuntime`.  

> I'm wondering if it makes sense to have types/enums follow a nested file module structure, so we can export * from "../types/enums/lambda"; in lambda/index.ts, which would allow us to maintain the aws.lambda.NodeJS10dXRuntime enum path.

This sounds right to me.  I think this just requires writing separate files for each module inside the `enums` folder.